### ISSUE TITLE
Class method mocks are not working properly.

### ISSUE THREAD

**Issue by achilles42** [waterlink/mocks.cr#2]
Code: application.cr file

```
class Application
  def self.exists?(message)
     "wolrd"
  end
end

Spec: applcation_spec.cr file
describe Application do
   it "should mock the class mthods" do
      allow(Application).to receive(self.exists?("hello")).and_return("world")
      Application.exists?("hello") 
   end
end
```

Throwing following Error: 

Error in ./spec/application_spec.cr:12: instantiating                                                                     

```
Error in ./spec/application_spec.cr:12: instantiating 'Mocks::Allow(Application:Class)#to(Mocks::Message)'

    allow(Application).to receive(self.exists?("hello")).and_return("world")
                                 ^~

in ./libs/mocks/mocks/allow.cr:16: instantiating 'Mocks::Registry#fetch_method(String)'

      Mocks::Registry
         ^~~~~~~~~~~~

in ./libs/mocks/mocks/registry.cr:24: instantiating 'Hash(String, Mocks::Registry::Method)#fetch(String)'

      methods[method_name] = methods.fetch(method_name) {
                                     ^~~~~

in ./libs/mocks/mocks/registry.cr:24: instantiating 'Hash(String, Mocks::Registry::Method)#fetch(String)'

      methods[method_name] = methods.fetch(method_name) {
                                     ^~~~~

in ./libs/mocks/mocks/registry.cr:25: instantiating 'Mocks::Registry::Method:Class#new()'

        Method.new
               ^~~

instantiating 'Mocks::Registry::Method#initialize()'

in ./libs/mocks/mocks/registry.cr:52: instantiating 'Mocks::Registry::Stubs:Class#new()'

        @stubs = Stubs.new
                       ^~~

instantiating 'Mocks::Registry::Stubs#initialize()'

in ./libs/mocks/mocks/registry.cr:118: can't use Mocks::Registry::Result(T) as generic type argument yet, use a more specific type

        @hash = {} of {ObjectId, Args} => Result
                                          ^~~~~~

```

**Comment by waterlink** [comment]
@spraveen1112 Thanks for the report! Could you tell which version of Crystal you use here? (`0.10.0` ?)

**Comment by achilles42** [comment]
@waterlink : yup, Crystal 0.10.0

**Comment by waterlink** [comment]
@spraveen1112 Everything works correctly for me. It feels like you use way outdated version of `mocks`. Maybe the only released version at that.

If so, it is my fault for not releasing any new version (I myself use `master` branch everywhere for `mocks`).

**Comment by waterlink** [comment]
And don't forget to actually define the mock itself: https://github.com/waterlink/mocks.cr/blob/master/spec/issues/2_spec.cr#L9-L11 - otherwise it is not gonna work, there is no way library can figure this information out with current state of the compiler (and probably never would be able to).

**Comment by waterlink** [comment]
@spraveen1112 I have released `0.2.0`, can you try adding `version: 0.2.0` to your `shard.yml`'s dependency for `mocks` and see if it works?

**Comment by achilles42** [comment]
@waterlink: thanks.... its working fine now :+1:
